src/features/klant/bedrijf/service.ts

```diff
@@ -122,11 +127,33 @@ const mapVestiging = ({
   };
 };
 
+const hasFoutCode = (body: unknown, code: string) => {
+  if (
+    body &&
+    typeof body === "object" &&
+    "fout" in body &&
+    Array.isArray(body.fout)
+  ) {
+    return body.fout.some((x) => x?.code === code);
+  }
+  return false;
+};
+
+export class FriendlyError extends Error {}
+
 function searchBedrijvenInHandelsRegister(url: string) {
-  return fetchLoggedIn(url)
-    .then(throwIfNotOk)
-    .then(parseJson)
-    .then(parseKvkPagination);
+  return fetchLoggedIn(url).then(async (r) => {
+    if (r.status === 404) {
+      const body = await r.json();
+      if (hasFoutCode(body, "IPD5200")) return defaultPagination([]);
```

Comment:
dus als het een 404 is en er staat niet 'fout' in de body, dan krijg je een standaard error. dit lijkt me fragiel. waarom niet altijd bij een 404 gewoon een leeg resultaat retourneren?

Reply:
404 zou ook kunnen betekenen dat we een verkeerd endpoint aanroepen. Als we bij alle 404 responses een leeg resultaat retourneren zou dat onopgemerkt blijven. De IPD code staat hier gedocumenteerd: https://developers.kvk.nl/documentation#status-codes